graph/encoding/dot/encode.go

```diff
@@ -510,10 +512,18 @@ func (p *multiGraphPrinter) print(g graph.Multigraph, name string, needsIndent,
 
 			for _, l := range lines {
 				lid := l.ID()
-				if p.visited[line{inGraph: name, id: lid}] {
-					continue
+				if isDirected {
+					if p.visited[line{inGraph: name, from: nid, to: tid, id: lid}] {
+						continue
+					}
+					p.visited[line{inGraph: name, from: nid, to: tid, id: lid}] = true
```

Comment:
perhaps construct that key once (before the `if` at line 516) and use in both places?

Reply:
Done